setup.py

```diff
@@ -188,7 +188,7 @@ def run(self):
         ],
         "sphinx_immaterial.apidoc.cpp.cppreference_data": ["*.xml"],
     },
-    python_requires=">=3.7",
+    python_requires=">=3.8",
```

Comment:
❓ python 3.7 is still getting security fixes until 2023-06-27. Is there a reason to drop 3.7 support?

Reply:
Sphinx 6 drops support for Python 3.7, and making the monkey patch work for 3.7 when using Sphinx < 6 would require even more branching.  But I see that I need to revise this anyway, so perhaps I can fix it.

Reply:
That's good enough reason for me (it just wasn't obvious).